### PR TITLE
Fix slow table

### PR DESCRIPTION
**Pull request by nvaytet**
The new implementation of `table` (#907) was very slow to render when a large number of rows was present in the data (~30s to render the table for 3M rows), even if only a small subset of the rows was being displayed.

This came from the use of an `SelectionSlider` whose array of `options` was being converted to strings.

This new implementation changes many things and the diff might be difficult to follow. At the same time as replacing the `SelectionSlider` with a `IntSlider`, we are now no longer copying `Variables` into a new `Dataset` when constructing the table columns. Instead, we store the `Variables` into a `dict` of `Variables`.

**Reviewer:** make sure to test on your machine with a few cases. Some slow-down is apparent when displaying ~200 rows and more, since the update of the table is now continuous with the slider movement (this is a biproduct of having to use a `IntSlider` which cannot be flipped to be top to bottom like a scrollbar, and therefore needs an additional `Label` widget to display the starting row index).

To test: try the following or a variant
```Py
d = sc.Dataset()
d.coords[Dim.Position] = sc.Variable([Dim.Position], values=np.arange(10.0), variances=np.random.rand(10))
d.coords[Dim.X] = sc.Variable([Dim.X], values=np.arange(11))
d.coords[Dim.Y] = sc.Variable([Dim.Y], values=np.arange(12))
d.labels["sample"] = sc.Variable([Dim.Position], values=np.arange(9))
d.labels["eee"] = sc.Variable([Dim.Position], values=np.arange(10))
d.masks["eee"] = sc.Variable([Dim.Position], values=np.zeros([10]))
d['sample'] = sc.Variable([Dim.Position], values=np.arange(10))
d['sample2'] = sc.Variable([Dim.Position], values=np.arange(9.0), variances=np.random.rand(9))
d['Xaa'] = sc.Variable([Dim.X], values=np.arange(10))
d['scalar'] = sc.Variable(1.5)
d['scalarAndVar'] = sc.Variable(21.5, variance=0.56)
d.labels["kkk"] = sc.Variable(91.5, variance=0.56)
sc.table(d)
sc.table(d["sample"])
sc.table(d["sample"].data)
sc.table(d["sample"].data.values)
```
and with a large number of rows:
```Py
d = sc.Dataset()
N = 20000
d.coords[Dim.Tof] = sc.Variable([Dim.Tof], values=np.arange(N).astype(np.float64),
                                unit=sc.units.us)
d.coords[Dim.X] = sc.Variable([Dim.X], values=np.arange(N).astype(np.float64),
                                unit=sc.units.m)
d['Sample'] = sc.Variable([Dim.Tof], values=10.0*np.random.rand(N),
                          unit=sc.units.counts, variances=np.random.rand(N))
d['Background'] = sc.Variable([Dim.Tof], values=10.0*np.random.rand(N-1),
                          unit=sc.units.counts)
d['hitme'] = sc.Variable([Dim.Tof], values=10.0*np.random.rand(N),
                          unit=sc.units.counts)
d['weight'] = sc.Variable([Dim.X], values=10.0*np.random.rand(N),
                          unit=sc.units.kg)
sc.table(d)
```

### CODE REVIEW

File: python/src/scipp/runtime_config.py [hunk 1]
```diff
@@ -77,11 +77,11 @@
...
     # The colors for each dataset member used in table and show functions
     "colors": {
-        "coord": "#dde9af",
+        "coords": "#dde9af",
         "data": "#ffe680",
         "labels": "#afdde9",
-        "attr": "#ff8080",
-        "mask": "#cccccc",
+        "attrs": "#ff8080",
+        "masks": "#d2b4de",
```
Comment (SimonHeybrock) [inline python/src/scipp/runtime_config.py]: Why the change to the inconsistent plural, and change of the color for masks?

File: python/src/scipp/table.py [hunk 2]
```diff
@@ -160,270 +128,196 @@ def table_from_dataset(dataset,
...
+                iterlist = scipp_obj.items()
+            else:
+                iterlist = [(scipp_obj.name, scipp_obj)]
+            # Note that capital 'D' is intentional on 'Data' to avoid the
+            # '.data' attribute of a DataArray being picked up
+            for field in ["coords", "labels", "masks", "Data"]:
+                if hasattr(scipp_obj, field):
+                    iter_items = getattr(scipp_obj, field).items()
+                else:
+                    iter_items = iterlist
```
Comment (SimonHeybrock) [inline python/src/scipp/table.py]:
```python
for tag, field in zip(["coord", "labels", "mask", "data"], [x.coords, x.labels, x.masks, iterlist]):
```
less hacky and easier to read?

**Comment by SimonHeybrock** [comment]
Tested locally, works, and is awesome! Much smoother and really quick, the immediate update (before releasing the slider) is also great.

File: python/src/scipp/runtime_config.py (hunk 1 above)
Comment (nvaytet) [reply python/src/scipp/runtime_config.py]: oh, the masks color i changed because it looked very similar to the headers in the table where it says `Values` and `Variances`, but it's not that important.

Changing to plural for masks and attrs makes sense to me, because it's like labels, there can be more than one, and the attributes of the `Dataset` are `.attrs` and `.masks`.
It was the same reason for the coords: the attribute is `.coords`, although one could argue that you will only have one coordinate in a table. However, you can have more than one in a `show` output.

Finally, it makes it possible to write very succint and generic code, however, it is now more inflexible since if you change this back, I can no longer use the `getattr`s [here](https://github.com/scipp/scipp/blob/f08345964be6266d2816a65c20d9f0b0f9aa058a/python/src/scipp/table.py#L217)

File: python/src/scipp/table.py (hunk 2 above)
Comment (nvaytet) [reply python/src/scipp/table.py]: yes this is better since it gets rid of the `getattr`

File: python/src/scipp/runtime_config.py (hunk 1 above)
Comment (SimonHeybrock) [reply python/src/scipp/runtime_config.py]: > Changing to plural for masks and attrs makes sense to me, because it's like labels, there can be more than one, and the attributes of the Dataset are .attrs and .masks.
It was the same reason for the coords: the attribute is .coords, although one could argue that you will only have one coordinate in a table. However, you can have more than one in a show output.

The attributes are all plural, because it returns the dict of coords, labels, masks, or attributes. When talking about an individual entry (see `setCoord`, `setLabels`, `setMask`, `setData`), only "labels" is plural, because for all the others the word collectively refers to an array (coord is an array of values, a mask is an array of boolean values, data is an array), whereas "the label is an array" is weird, so this one has the plural "s". If anything, I would consider removing the `s` from labels, everywhere.

Comment (nvaytet) [reply python/src/scipp/runtime_config.py]: you could also just think of it as "this is the color for masks", instead of "this is the color for a mask" and it would still make sense.
I understand the point about `setMask` but then `setData` is also inconsistent, being a plural word.
Anyway, these colors are only used on the python side, so I thought it made sense because a python user will never see the C++ side of things.

Comment (SimonHeybrock) [reply python/src/scipp/runtime_config.py]: I'm ok with that, but could you change the masks back to grey? I feel that is more intuitive, e.g., visually similar to grey-out GUI elements or rows in a table.
